README.md
# EDLIN

EDLIN line editor written for FreeDOS

http://www.ibiblio.org/pub/micro/pc-stuff/freedos/files/dos/edlin/2.21/



ChangeLog
2022-06-25 Gregory Pietsch <gpietsch@comcast.net>
	* ow.bat : Changed version to 2.21.
	* Released version 2.21.

2022-06-19 Gregory Pietsch <gpietsch@comcast.net>
	* edlib.c (copy_block) : Fixed so that it works correctly.
	* edlin.c : Fixed defaults for 'C' (copy).
	* configure.ac : Bumped version to 2.21.

2022-06-04 Gregory Pietsch <gpietsch@comcast.net>
	* edlib.c (insert_block) : Changed so that a number off the end of
	  the buffer just appends to the buffer instead of giving an error
	  message. This matches historical behavior.

2022-05-15 Gregory Pietsch <gpietsch@comcast.net>
	* edlin.en : Copied out of kitten directory.
	* kitten/* : Removed. Jerome Shibel's NLS is taking over. 
	* msgs-*.h : Removed. See above.
	* edlin-ja.htm edlin-ja.txt : Removed. See above.
	* edlin.c : Added '#' as a synonym for "$+1".
	* edlin.en msgs.h msgs-en.h msgscats.h : Added "New file." message.
	* edlib.c edlib.h (file_exists) : Made non-static and added
	  prototype to header file.
	* edlin.c : Added "New file." for new filenames on command line.
	* catgets.c defines.c defines.h dynstr.c edlib.c nl_types.h : 
	  Removed obsolete STDC_HEADERS check. The program
	  should be able to include all C89/C90 Standard C headers 
	  unconditionally.
	* edlib.c edlib.h (quitting) : New function to verify quitting.
	* edlin.c : Calls quitting on the 'q' and 'e' functions to
	  verify if the user really wants to quit.
	* edlin.en msgs.h msgs-en.h msgscats.h : 
	  Added "Abort edit (Y/N)? " message.
	* configure.ac : Removed obsolete STDC_HEADERS check.
	* configure.ac : Bumped version to 2.20.
	* Makefile.am : Removed various items such as kitten directory
	  and msgs-*.h files. The msgs-en.h and edlin.en files will be
	  retained as seeds for the NLS.
	* edlin.htm : Modified to reflect new behaviors.
	* ow.bat : Bumped version to 2.20.
	* Version 2.20 released.

2021-03-21 Gregory Pietsch <gpietsch@comcast.net>
	* edlib.c (read_line) : Fixed so that EOF returns null pointer.
	* edlib.c (insert_block) : EOF on input now just aborts to command
	  prompt.
	* edlin.c (main) : EOF on input just aborts. Is there something better
	  to do?
	* configure.ac : Version bumped to 2.19.
	* config-h.bc config-h.ow : Version bumped to 2.19.
	* Version 2.19 released.

2020-03-22 Gregory Pietsch <gpietsch@comcast.net>
	* catgets.c : Reprogrammed catopen function.
	* catgets.c : Added strrchr implementation.
	* edlin.1.gz : New file, from Darren Goosens.
	* Makefile.am : Added line for edlin.1.gz.
	* configure.ac : Added test for strrchr.
	* Version 2.18 released.

2017-11-30 Gregory Pietsch <gpietsch@comcast.net>
	* configure.ac : Redone from a configure.scan. Some old unneeded
          tests were removed.
	* config-h.ow config-h.bc : Regenerated to remove a few unneeded tests.
	* Version 2.17 released.

2017-11-29 Gregory Pietsch <gpietsch@comcast.net>
	* configure.ac : Renamed from configure.in.
	* edlib.c (transfer_file) : Fixed problem found by Axel Scheepers.
	  This was regarding transferring a non-existent file.

2017-08-21 Gregory Pietsch <gpietsch@comcast.net>
	* msgs-sl.h kitten/edlin.sl : New Slovenian translation from FreeDOS
	  internationalization project.
	* msgs-tr.h kitten/edlin.tr : New Turkish translation from FreeDOS
	  internationalization project.
	* edlib.c : added #include <process.h> for Borland.
	* configure.in : added test for process.h and changed version number.
	* AUTHORS : added translators for Slovenian and Turkish.
	* config-h.bc config-h.ow : Added HAVE_PROCESS_H test and changed
	  version number.
	* ow.bat : Changed version number.

2010-12-20 Gregory Pietsch <gpietsch@comcast.net>
	* catgets.c : Fixed problems found by rugxulo.
	* ow.bat : Changed version number.
	* configure.in : Changed version number.
	* config-h.bc config-h.ow : Changed version number.
	* Version 2.15 released.

2010-09-25 Gregory Pietsch <gpietsch@comcast.net>
	* edlib.c : New version of strchr.
	* catgets.c : Added new version of strchr here.
	* catgets.c : Added append_from_file and fixed cat_read so that it
	  reads a line at a time instead of going character by character.
	* ow.bat : Changed version number.

2010-03-02 Gregory Pietsch <gpietsch@comcast.net>
	* edlib.c : Added read_from_file() and fixed transfer_file() to call
	  it instead of reading character by character.  The
	  read_from_file() function calls fgets() to get lines from the
	  file.
	* config-h.bc config-h.ow : Changed version number.
	* ow.bat : Changed version number.
	* edlin-ja.htm edlin.htm : Changed version number.

2009-07-27 Gregory Pietsch <gpietsch@comcast.net>
	* catgets.c : Bug fixes. Bugs originally found by rugxulo.
	* AUTHORS : Added rugxulo (author of ow.bat).
	* ow.bat : Fixed rugxulo's e-mail address.
	* Version 2.12a released.

2009-07-24 Gregory Pietsch <gpietsch@comcast.net>
	* edlib.c : Minor bug fix.
	* Version 2.12 released.

2009-02-11 Gregory Pietsch <gpietsch@comcast.net>
	* Version 2.11 released.

2008-08-17 Gregory Pietsch <gpietsch@comcast.net>
        * Minor fixes to comments in dynarray.h, dynstr.c.
        * Added Tstorage_class to dynarray.h to make exported functions
          static.
        * catgets.c : Made catopen() more robust.

2007-06-02 Gregory Pietsch <gpietsch@comcast.net>
	* Fixed spelling mistakes -- grrr!
	* ow.bat : changed to reflect new version.
	* config-h.ow config-h.bc : regenerated.
	* Version 2.10c released.

2007-05-29 Gregory Pietsch <gpietsch@comcast.net>
	* kitten/edlin.eo : New file - A translation of the message strings
	  into Esperanto. Thanks, rugxulo!
	* msgs-eo.h : New file - generated from kitten/edlin.eo.
	* ow.bat : changed to reflect new directory.
	* Version 2.10b released.

2007-05-28 Gregory Pietsch <gpietsch@comcast.net>
	* dynarray.h : debugged. Grrrr....
	* ow.bat : Replaced with new version.
	* Version 2.10a released.

2007-05-26 Gregory Pietsch <gpietsch@comcast.net>
	* kitten/edlin.la : New file - A translation into Latin.

msgscats.h
/* msgs.h - messages file for catgets()

   Author: Gregory Pietsch

   DESCRIPTION:

   This file contains #defines for all the message strings in edlin.
   For internationalization fun, just translate the messages in this
   file.

   The strings were generated using ESR's cstrings program and moved
   here.

*/

#ifndef MSGS_H
#define MSGS_H

#ifdef USE_KITTEN
#include "kitten.h"
#else
#include "nl_types.h"
#endif

#define YES     catgets(the_cat, 1, 0, "Yy")

#define G00001  catgets(the_cat, 1, 1, " : ")
#define G00002  catgets(the_cat, 1, 2, "O.K.? ")
#define G00003  catgets(the_cat, 1, 3, "Entry error")
#define G00004  catgets(the_cat, 1, 4, "%s: %lu line read\n")
#define G00005  catgets(the_cat, 1, 5, "%s: %lu lines read\n")
#define G00006  catgets(the_cat, 1, 6, "%s: %lu line written\n")
#define G00007  catgets(the_cat, 1, 7, "%s: %lu lines written\n")
#define G00008  catgets(the_cat, 1, 8, "%lu:%c%s\n")
#define G00009  catgets(the_cat, 1, 9, "Press <Enter> to continue")
#define G00010  catgets(the_cat, 1, 10, "%lu: ")
#define G00011  catgets(the_cat, 1, 11, "Not found")
#define G00012  catgets(the_cat, 1, 12, "%lu: %s\n")
#define G00013  catgets(the_cat, 1, 13, "\nedlin has the following subcommands:\n")
#define G00014  catgets(the_cat, 1, 14, "#                 edit a single line    [#],[#],#m        move")
#define G00015  catgets(the_cat, 1, 15, "a                 append                [#][,#]p          page")
#define G00016  catgets(the_cat, 1, 16, "[#],[#],#,[#]c    copy                  q                 quit")
#define G00017  catgets(the_cat, 1, 17, "[#][,#]d          delete                [#][,#][?]r$,$    replace")
#define G00018  catgets(the_cat, 1, 18, "e<>               end (write & quit)    [#][,#][?]s$      search")
#define G00019  catgets(the_cat, 1, 19, "[#]i              insert                [#]t<>            transfer")
#define G00020  catgets(the_cat, 1, 20, "[#][,#]l          list                  [#]w<>            write\n")
#define G00021  catgets(the_cat, 1, 21, "where $ above is a string, <> is a filename,")
#define G00022  catgets(the_cat, 1, 22, "# is a number (which may be .=current line, $=last line,")
#define G00023  catgets(the_cat, 1, 23, "or either number + or - another number).\n")
#define G00024  catgets(the_cat, 1, 24, ", copyright (c) 2003 Gregory Pietsch")
#define G00025  catgets(the_cat, 1, 25, "This program comes with ABSOLUTELY NO WARRANTY.")
#define G00026  catgets(the_cat, 1, 26, "It is free software, and you are welcome to redistribute it")
#define G00027  catgets(the_cat, 1, 27, "under the terms of the GNU General Public License -- either")
#define G00028  catgets(the_cat, 1, 28, "version 2 of the license, or, at your option, any later")
#define G00029  catgets(the_cat, 1, 29, "version.\n")
#define G00030  catgets(the_cat, 1, 30, "Out of memory")
#define G00031  catgets(the_cat, 1, 31, "String length error")
#define G00032  catgets(the_cat, 1, 32, "String position error")
#define G00033  catgets(the_cat, 1, 33, "Invalid user input, use ? for help.")
#define G00034  catgets(the_cat, 1, 34, "No filename")
#define G00035  catgets(the_cat, 1, 35, "Buffer too big")
#define G00036  catgets(the_cat, 1, 36, "Invalid buffer position")
#define G00037  catgets(the_cat, 1, 37, "ERROR: %s\n")
#define G00038	catgets(the_cat, 1, 38, "New file.")
#define G00039	catgets(the_cat, 1, 39, "Abort edit (Y/N)? ")


#ifndef EXTERN
#define EXTERN extern
#endif
EXTERN nl_catd the_cat;

#endif

/* END OF FILE */
